presto-spark-base/src/main/java/com/facebook/presto/spark/execution/property/WorkerProperty.java

```diff
@@ -22,17 +22,87 @@
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
 
 /**
- * A utility class that helps with properties and its materialization.
+ * The implementor of any new WorkerProperty is expected to provide three configuration classes (or their variants) as minimal requirements to run the native process.
+ * The variants of the new configuration classes need to extend these three as base class to provide any new configs or override existing configs. For example:
+ * <p>
+ * // new configuration variance
+ * public class InternalNativeExecutionNodeConfig
+ * extends NativeExecutionNodeConfig
+ * {
+ * private static final String SMC_SERVICE_INVENTORY_TIER = "smc-service-inventory.tier";
+ * <p>
+ * private String smcServiceInventoryTier = "dummy.tier";
+ *
+ * @Override public Map<String, String> getAllProperties()
+ * {
+ * ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+ * return builder.putAll(super.getAllProperties())
+ * .put(SMC_SERVICE_INVENTORY_TIER, smcServiceInventoryTier)
+ * .build();
+ * }
+ * @Config(SMC_SERVICE_INVENTORY_TIER) public PrestoFacebookSparkNativeExecutionNodeConfig setSmcServiceInventoryTier(String smcServiceInventoryTier)
+ * {
+ * this.smcServiceInventoryTier = smcServiceInventoryTier;
+ * return this;
+ * }
+ * }
+ * <p>
+ * // New WorkerProperty with new InternalNativeExecutionNodeConfig
+ * public class InternalWorkerProperty
+ * extends WorkerProperty<NativeExecutionConnectorConfig, InternalNativeExecutionNodeConfig, NativeExecutionSystemConfig>
+ * {
+ * @Inject public InternalWorkerProperty(
+ * NativeExecutionSystemConfig systemConfig,
+ * NativeExecutionConnectorConfig connectorConfig,
+ * InternalNativeExecutionNodeConfig nodeConfig)
+ * {
+ * super(connectorConfig, nodeConfig, systemConfig);
+ * }
+ * }
  */
-public class WorkerProperty
+public class WorkerProperty<T1 extends NativeExecutionConnectorConfig, T2 extends NativeExecutionNodeConfig, T3 extends NativeExecutionSystemConfig>
 {
-    private WorkerProperty()
+    private final T1 connectorConfig;
+    private final T2 nodeConfig;
+    private final T3 systemConfig;
+
+    public WorkerProperty(
+            T1 connectorConfig,
+            T2 nodeConfig,
+            T3 systemConfig)
+    {
+        this.systemConfig = requireNonNull(systemConfig, "systemConfig is null");
+        this.nodeConfig = requireNonNull(nodeConfig, "nodeConfig is null");
+        this.connectorConfig = requireNonNull(connectorConfig, "connectorConfig is null");
+    }
+
+    public T3 getSystemConfig()
+    {
+        return systemConfig;
+    }
+
+    public T2 getNodeConfig()
+    {
+        return nodeConfig;
+    }
+
+    public T1 getConnectorConfig()
+    {
+        return connectorConfig;
+    }
```

Comment:
Define the methods in same order as T1, T2 and T3 return types.

Reply:
Done.